src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -854,5 +854,24 @@ public static bool IsEnumType(this ITypeSymbol type)
         {
             return type.IsValueType && type.TypeKind == TypeKind.Enum;
         }
+
+        public static bool IsMutableValueType(this ITypeSymbol type)
+        {
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                return false;
+            }
+
+            foreach (var member in type.GetMembers())
+            {
+                if (member is IFieldSymbol fieldSymbol &&
+                    !(fieldSymbol.IsConst || fieldSymbol.IsReadOnly || fieldSymbol.IsStatic))
+                {
+                    return true;
+                }
+            }
```

Comment:
i think we should whitelist some structs here.  otherwise, common patterns will break.

Reply:
Note: handling Nullable specially doesn't even seem like a problem to me given it's a core language concept.

Reply:
If we add special handling for `Nullable<T>`, it would be for cases where `!typeof(T).IsMutableValueType()`, which does cover a majority of relevant cases.

Reply:
That would be fine with me :)